core/domain/learner_progress_services.py

```diff
@@ -868,6 +1322,93 @@ def _get_filtered_collection_playlist_summaries(
         nonexistent_playlist_collection_ids)
 
 
+def get_story_summary_dicts(user_id, story_summaries):
+    """Returns a displayable summary dict of the the story summaries
+    given to it.
+
+    Args:
+        user_id: str. The id of the learner.
+        story_summaries: list(StorySummary). A list of the
+            summary domain objects.
+
+    Returns:
+        list(dict). The summary dict objects corresponding to the given summary
+        domain objects.
+    """
+    summary_dicts = []
+    for story_summary in story_summaries:
+        story = story_fetchers.get_story_by_id(story_summary.id)
+        topic = topic_fetchers.get_topic_by_id(story.corresponding_topic_id)
```

Comment:
Use get_multi outside, instead of get() in a for loop. Having said that, does this summary have more fields that in StorySummary object? If so, rename it to `displayable_story_summary_dicts` or something that like that.

Also, this is only called from the controller, right (add it to docstring, if so)? Since, we shouldn't be passing dicts around within the backend. If it is only from the controller, it is fine, since that has to be returned to the frontend anyway. if not, you need to create a domain object for this dict and return that.